File: src/Entity/Promotion.php
<?php

namespace App\Entity;

use App\Repository\PromotionRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=PromotionRepository::class)
 */
class Promotion
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="date")
     */
    private $dated;

    /**
     * @ORM\Column(type="date")
     */
    private $datef;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $description;

    /**
     * @ORM\Column(type="integer")
     */
    private $pourcentage;
    /**
     * @ORM\Column(type="integer")
     */
    private $vote;


    /**
     * @ORM\ManyToOne(targetEntity=Codecoupone::class, inversedBy="promotion")
     */
    private $codecoupone;

    /**
     * @ORM\OneToMany(targetEntity=Postlike::class, mappedBy="post")
     */
    private $likes;

    public function __construct()
    {
        $this->likes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }
    public function getDated(): ?\DateTimeInterface
    {
        return $this->dated;
    }

    public function setDated(\DateTimeInterface $dated): self
    {
        $this->dated = $dated;

        return $this;
    }

    public function getDatef(): ?\DateTimeInterface
    {
        return $this->datef;
    }

    public function setDatef(\DateTimeInterface $datef): self
    {
        $this->datef = $datef;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }





    public function getPourcentage(): ?int
    {
        return $this->pourcentage;
    }

    public function setPourcentage(int $pourcentage): self
    {
        $this->pourcentage = $pourcentage;

        return $this;
    }
    public function getVote(): ?int
    {
        return $this->vote;
    }

    public function setVote(int $vote): self
    {
        $this->vote = $vote;

        return $this;
    }


    public function getCodecoupone(): ?Codecoupone
    {
        return $this->codecoupone;
    }

    public function setCodecoupone(?Codecoupone $codecoupone): self
    {
        $this->codecoupone = $codecoupone;

        return $this;
    }

    public function __toString()
    {
        return $this->Codecoupone;
    }

    /**
     * @return Collection<int, Postlike>
     */
    public function getLikes(): Collection
    {
        return $this->likes;
    }
    public function setLikes(?Likes $likes): self
    {
        $this->likes = $likes;

        return $this;
    }


    public function addLike(Postlike $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setPost($this);
        }

        return $this;
    }

    public function removeLike(Postlike $like): self
    {
        if ($this->likes->removeElement($like)) {
            // set the owning side to null (unless already changed)
            if ($like->getPost() === $this) {
                $like->setPost(null);
            }
        }

        return $this;
    }

}

File: codename-project/src/com/mycomany/entities/Promotion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycomany.entities;

import java.util.Date;

/**
 *
 * @author PC
 */
public class Promotion {
    
    private int id;
    private String description;
    private String dated;
    private String datef;
    private int pourcentage;
    private int codecoupone;
    public Promotion(){
        
    }
  
    public Promotion(int id,int pourcentage, String description , String dated, String datef){
        this.id=id;
        this.dated=dated;
        this.datef=datef;
        this.description=description;
        this.pourcentage=pourcentage;
    }
     public Promotion(String description , String dated, String datef,int pourcentage){
     
        this.dated=dated;
        this.datef=datef;
        this.description=description;
        this.pourcentage=pourcentage;
    }

    public String getDated() {
        return dated;
    }

    public String getDatef() {
        return datef;
    }

    public String getDescription() {
        return description;
    }

    public int getId() {
        return id;
    }

    public int getPourcentage() {
        return pourcentage;
    }

    public void setDated(String dated) {
        this.dated = dated;
    }

    public void setDatef(String datef) {
        this.datef = datef;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setPourcentage(int pourcentage) {
        this.pourcentage = pourcentage;
    }

    public int getCodecoupone() {
        return codecoupone;
    }

    public void setCodecoupone(int codecoupone) {
        this.codecoupone = codecoupone;
    }

    public void setPourcentage(String pourcentage) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
     
}
